tomopy/misc/morph.py

```diff
@@ -121,15 +125,56 @@ def pad(arr, axis, npad=None, mode='constant', **kwargs):
     if npad is None:
         npad = _get_npad(arr.shape[axis])
 
-    pad_width = _get_pad_sequence(arr.shape, axis, npad)
-
-    return np.pad(arr, pad_width, str(mode), **kwargs)
+    newshape = list(arr.shape)
+    newshape[axis] += 2*npad
+
+    slc_in, slc_l, slc_r, slc_l_v, slc_r_v = _get_slices(arr.shape, axis, npad)
+
+    out = np.empty(newshape, dtype=arr.dtype)
+    if arr.dtype in [np.float32, np.float64, np.bool,
+                     np.int32, np.int64, np.complex128]:
+        # Datatype supported by numexpr
+        with mproc.set_numexpr_threads(ncore):
+            ne.evaluate("arr", out=out[slc_in])
+            if mode == 'constant':
+                np_cast = getattr(np, str(arr.dtype))
+                cval = np_cast(kwargs['constant_values'])
+                ne.evaluate("cval", out=out[slc_l])
+                ne.evaluate("cval", out=out[slc_r])
+            else:
+                ne.evaluate("vec", local_dict={'vec': arr[slc_l_v]},
+                            out=out[slc_l])
+                ne.evaluate("vec", local_dict={'vec': arr[slc_r_v]},
+                            out=out[slc_r])
+    else:
+        # Datatype not supported by numexpr, use numpy instead
+        out[slc_in] = arr
+        if mode == 'constant':
+            out[slc_l] = kwargs['constant_values']
+            out[slc_r] = kwargs['constant_values']
+        else:
```

Comment:
perhaps it will be more clear to have an `mode == 'edge'` and raise an error if kwarg `mode` is not constant, edge or whatever else is allowed.